src/Avalonia.Base/Rendering/Composition/CompositingRenderer.cs

```diff
@@ -299,7 +294,7 @@ public void Dispose()
         // Wait for the composition batch to be applied and rendered to guarantee that
         // render target is not used anymore and can be safely disposed
         if (Compositor.Loop.RunsInBackground)
-            _compositor.RequestCommitAsync().Wait();
+            _compositor.Commit().Wait();
```

Comment:
The app can hang here if a popup is closed after a very short time.
In my case happens with AvaloniaEdit completionwindow